### ISSUE TITLE
1.0.0.pre.rc2 "SimpleCov failed to recognize the test framework" error

### ISSUE THREAD

**Issue by rowlando** [codeclimate/ruby-test-reporter#135]
I am using SimpleCov 0.12.0 successfully to generate coverage reports. I configure SimpleCov using a `.simplecov` file:

```
SimpleCov.formatters  = [SimpleCov::Formatter::HTMLFormatter]
SimpleCov.start
```

`spec/spec_helper.rb` and `features/support/env.rb` both require `require 'simplecov'` at the top of their respective files.

When I run `bundle exec rake` the following lines are included in the console output:

``` bash
Coverage report generated for RSpec to /Users/nickrowlands/Projects/mediators/family-mediators-api/coverage. 163 / 166 LOC (98.19%) covered.
Coverage report generated for Cucumber Features, RSpec to /Users/nickrowlands/Projects/mediators/family-mediators-api/coverage. 168 / 170 LOC (98.82%) covered.
```

My Gemfile includes the following line:

```
gem 'codeclimate-test-reporter', '1.0.0.pre.rc2', require: nil
```

When I run the command:

``` bash
CODECLIMATE_REPO_TOKEN=***** bundle exec codeclimate-test-reporter
```

the following message is displayed in the console:

``` bash
I, [2016-10-21T12:21:11.183457 #33664]  INFO -- : Reporting coverage data to Code Climate.
SimpleCov failed to recognize the test framework and/or suite used. Please specify manually using SimpleCov.command_name 'Unit Tests'.
Sending report to https://codeclimate.com for branch ci_code_climate... done.
```

I can't work out if I'm doing something wrong or if there's a bug.

**Comment by matchbookmac** [comment]
Running into the same issue.

**Comment by maxjacobson** [comment]
Thanks for opening this issue.

[SimpleCov prints this warning](https://github.com/colszowka/simplecov/blob/6e5ec2342cc1cfea16ca6c3bb1034062267defbf/lib/simplecov/command_guesser.rb#L43-L56) when it's running in a Ruby context that doesn't have specific constants defined. We used to (pre 1.0.0) run our test reporter within the context of our tests, so those constants would be defined. We now run the reporter as a separate command, so it can't rely on them being defined.

I see that coverage [is showing up for you](https://codeclimate.com/github/ministryofjustice/family-mediators-api/coverage), so I think things are generally working, but this output is undesirable. I traced it down to [this line](https://github.com/codeclimate/ruby-test-reporter/blob/ccbc87a44e1e235e8c28ea2f974dc87c405c1d1b/lib/code_climate/test_reporter/formatter.rb#L74) which just passes the test framework along in the test report. It seems like we no longer have access to that information as a consequence of running the reporter separate from the tests.

I'll talk to some colleagues about whether we actually need that information and get back to you. Cheers 👍
